source/Camera.cpp

```diff
@@ -0,0 +1,84 @@
+/* Camera.cpp
+Copyright (c) 2025 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Camera.h"
+
+#include "Preferences.h"
+
+#include <cmath>
+
+using namespace std;
+
+namespace {
+	const double CAMERA_VELOCITY_TRACKING = 0.1;
+	const double CAMERA_POSITION_CENTERING = 0.01;
+}
+
+
+
+void Camera::SnapTo(const Point &target)
+{
+	oldTarget = target;
+	center = target;
```

Comment:
```suggestion
void Camera::SnapTo(const Point &target, bool keepVelocity)
{
	// If a player is jumping to a new system, preserve their velocity by offsetting oldTarget and center.
	Point newTarget = keepVelocity ? target - velocity : target;
	oldTarget = newTarget;
	center = newTarget;

```

Reply:
Adding an extra line to explain how this works to achieve the goal when velocity is being zeroed out.
```suggestion
void Camera::SnapTo(const Point &target, bool keepVelocity)
{
	// If a player is jumping to a new system, preserve their velocity by offsetting oldTarget and center.
	// The correct velocity will be re-calculated when MoveTo is called later in this frame.
	Point newTarget = keepVelocity ? target - velocity : target;
	oldTarget = target;
	center = target;
```